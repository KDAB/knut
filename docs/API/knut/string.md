# String

Description of a RC file string. [More...](#detailed-description)

```qml
import Knut
```

## Properties

| | Name |
|-|-|
|string|**[id](#id)**|
|string|**[text](#text)**|

## Property Documentation

#### <a name="id"></a>string **id**

This property holds the id of the string.

#### <a name="text"></a>string **text**

This property holds the text of the string.
