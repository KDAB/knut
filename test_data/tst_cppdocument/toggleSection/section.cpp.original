#include "section.h"

Section::Section()
{
    int i = 0;
    ++i;
}

Section::~Section()
{
    int i = 0;
    ++i;
}

void Section::foo()
{
#ifdef KDAB_TEMPORARILY_REMOVED
    int i = 0;
    ++i;
#else // KDAB_TEMPORARILY_REMOVED
    qDebug("Section::foo is commented out");
    return;
#endif // KDAB_TEMPORARILY_REMOVED
}

int Section::bar()
{
    int i = 0;
    ++i;
    return i;
}

static std::string computeText()
{
    int i = 0;
    ++i;
    return "Hello World!";
}

Section *Section::create()
{
    int i = 0;
    ++i;
    computeText();
    return new Section;
}

Section &Section::reference()
{
    computeText();
    return *m_pointer;
}
