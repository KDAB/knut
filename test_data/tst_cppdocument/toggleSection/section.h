#pragma once

#include <string>

class Section {
public:
    Section();
    ~Section();
    void foo();
    int bar();

    static Section *create();
    Section &reference();
private:
    Section *m_pointer;
};
