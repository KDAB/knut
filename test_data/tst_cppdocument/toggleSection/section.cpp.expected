#include "section.h"

Section::Section()
{
#ifdef KDAB_TEMPORARILY_REMOVED
    int i = 0;
    ++i;
#else // KDAB_TEMPORARILY_REMOVED
    qDebug("Section::Section is commented out");
    return;
#endif // KDAB_TEMPORARILY_REMOVED
}

Section::~Section()
{
#ifdef KDAB_TEMPORARILY_REMOVED
    int i = 0;
    ++i;
#else // KDAB_TEMPORARILY_REMOVED
    qDebug("Section::~Section is commented out");
    return;
#endif // KDAB_TEMPORARILY_REMOVED
}

void Section::foo()
{
    int i = 0;
    ++i;
}

int Section::bar()
{
#ifdef KDAB_TEMPORARILY_REMOVED
    int i = 0;
    ++i;
    return i;
#else // KDAB_TEMPORARILY_REMOVED
    qDebug("Section::bar is commented out");
    return -1;
#endif // KDAB_TEMPORARILY_REMOVED
}

static std::string computeText()
{
#ifdef KDAB_TEMPORARILY_REMOVED
    int i = 0;
    ++i;
    return "Hello World!";
#else // KDAB_TEMPORARILY_REMOVED
    qDebug("computeText is commented out");
    return {};
#endif // KDAB_TEMPORARILY_REMOVED
}

Section *Section::create()
{
#ifdef KDAB_TEMPORARILY_REMOVED
    int i = 0;
    ++i;
    computeText();
    return new Section;
#else // KDAB_TEMPORARILY_REMOVED
    qDebug("Section::create is commented out");
    return nullptr;
#endif // KDAB_TEMPORARILY_REMOVED
}

Section &Section::reference()
{
#ifdef KDAB_TEMPORARILY_REMOVED
    computeText();
    return *m_pointer;
#else // KDAB_TEMPORARILY_REMOVED
    qDebug("Section::reference is commented out");
    return {};
#endif // KDAB_TEMPORARILY_REMOVED
}
